pytradfri/smart_task.py

```diff
@@ -273,17 +275,24 @@ def __repr__(self):
 class StartActionItemController:
     """Class to edit settings for a task."""
 
-    def __init__(self, item, raw, state, path, devices_dict):
-        """Initialize TaskControl."""
+    def __init__(
+        self,
+        item: StartActionItem,
+        raw: TypeRaw,
+        state: int,
```

Comment:
There seem to be some ambiguity about the state type as there are two different origins of `StartActionItem`. For one state is a boolean and for the other it's an integer. A boolean is a subtype of an integer but we may need to convert it to a number if we expect that.

Reply:
Yes, this is a horrible mess (I don't blame anyone, I wrote this code when I just got into Python).

I believe we can drop this property as I can't see it being used anywhere?
https://github.com/home-assistant-libs/pytradfri/blob/421a9237a0ef231f18aceb79320595957d28fe8d/pytradfri/smart_task.py#L203

That will remove one of the places `StartActionItem` is being used.

Reply:
`state` is normally a `bool` throughout the code so I went for that option and converted them to an `int` where needed.